### PR TITLE
[GLUTEN-10056][FLINK] Support function split_index

### PR DESCRIPTION
**Pull request by KevinyhZou**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

(Fixes: \#10056)

depends on pr: https://github.com/bigo-sg/velox/pull/8

## How was this patch tested?
UT

### CODE REVIEW

File: gluten-flink/planner/src/main/java/org/apache/gluten/rexnode/functions/SplitIndexRexCallConverter.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    if (!TypeUtils.isStringType(RexNodeConverter.toType(firstOp.getType()))
+        || !TypeUtils.isStringType(RexNodeConverter.toType(secondOp.getType()))
+        || !TypeUtils.isIntegerType(RexNodeConverter.toType(thirdOp.getType()))) {
+      return false;
+    }
+    if (!(secondOp instanceof RexLiteral) || !(thirdOp instanceof RexLiteral)) {
+      return false;
+    }
+    return true;
+  }
```
Comment (lgbo-ustc) [inline gluten-flink/planner/src/main/java/org/apache/gluten/rexnode/functions/SplitIndexRexCallConverter.java]: I think we could return true here without check, since it's the only convertor for `split_index`. The velox backend will do some validation check.

Comment (KevinyhZou) [reply gluten-flink/planner/src/main/java/org/apache/gluten/rexnode/functions/SplitIndexRexCallConverter.java]: fixed.

File: gluten-flink/planner/src/main/java/org/apache/gluten/rexnode/functions/SplitIndexRexCallConverter.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+  public SplitIndexRexCallConverter() {
+    super(FUNCTION_NAME);
+  }
+
+  @Override
+  public TypedExpr toTypedExpr(RexCall callNode, RexConversionContext context) {
+    List<TypedExpr> params = getParams(callNode, context);
+    ConstantTypedExpr indexExpr = (ConstantTypedExpr) params.get(params.size() - 1);
+    if (TypeUtils.isIntegerType(indexExpr.getReturnType())) {
```
Comment (shuai-xu) [inline gluten-flink/planner/src/main/java/org/apache/gluten/rexnode/functions/SplitIndexRexCallConverter.java:61]: Why need this transform? if it's not int, need to check it valid ?

File: gluten-flink/ut/src/test/java/org/apache/gluten/table/runtime/stream/custom/ScalarFunctionsTest.java [hunk 3]
```diff
@@ -126,4 +126,20 @@ void testEqual() {
     String query4 = "select c = d as x from tblEqual where a > 0";
     runAndCheck(query4, Arrays.asList("+I[false]", "+I[true]", "+I[false]"));
   }
+
+  @Test
+  void testSplitIndex() {
+    List<Row> rows =
+        Arrays.asList(
```
Comment (shuai-xu) [inline gluten-flink/ut/src/test/java/org/apache/gluten/table/runtime/stream/custom/ScalarFunctionsTest.java:135]: Is there any corner case need to test to verify the result same with flink?

**Comment by KevinyhZou** [comment]
> Which nexmark query use this function?

nexmark q22 need this

File: gluten-flink/planner/src/main/java/org/apache/gluten/rexnode/functions/SplitIndexRexCallConverter.java (hunk 2 above)
Comment (KevinyhZou) [reply gluten-flink/planner/src/main/java/org/apache/gluten/rexnode/functions/SplitIndexRexCallConverter.java:61]: The corresponding function in velox is `split_part`, and the last parameter's type is `int64_t`, but in flink `split_index` function, it is `integer` type, so we need this transform

File: gluten-flink/ut/src/test/java/org/apache/gluten/table/runtime/stream/custom/ScalarFunctionsTest.java (hunk 3 above)
Comment (KevinyhZou) [reply gluten-flink/ut/src/test/java/org/apache/gluten/table/runtime/stream/custom/ScalarFunctionsTest.java:135]: we can cover this by use ut in flink `ScalaFunctionsTest#testSplitIndex`

File: gluten-flink/planner/src/main/java/org/apache/gluten/rexnode/functions/SplitIndexRexCallConverter.java (hunk 2 above)
Comment (shuai-xu) [reply gluten-flink/planner/src/main/java/org/apache/gluten/rexnode/functions/SplitIndexRexCallConverter.java:61]: please add comments for it.

File: gluten-flink/ut/src/test/java/org/apache/gluten/table/runtime/stream/custom/ScalarFunctionsTest.java (hunk 3 above)
Comment (shuai-xu) [reply gluten-flink/ut/src/test/java/org/apache/gluten/table/runtime/stream/custom/ScalarFunctionsTest.java:135]: seems there are many null related tests in flink not covered by this test.

Comment (shuai-xu) [reply gluten-flink/ut/src/test/java/org/apache/gluten/table/runtime/stream/custom/ScalarFunctionsTest.java:135]: and others such as index is invalid.

Comment (KevinyhZou) [reply gluten-flink/ut/src/test/java/org/apache/gluten/table/runtime/stream/custom/ScalarFunctionsTest.java:135]: I have copyed some related flink tests here related to the index invalid, input has nulls, or delimiter is defined by ascii, but others tests e.g. `index` parameter is null or `delimiter` parameter is null, can not run here. as we can not pass the the null constant parameter value to velox. Or we should introduce a Variant to pass `null` constant.
